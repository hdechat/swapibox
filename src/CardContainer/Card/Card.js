import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({ card }) => {
  return (
    <article>
     <h2>{card.name}</h2>
     <h5>Species: {card.species}</h5>
     <h5>Homeworld: {card.homeworld}</h5>
     <p>pop: {card.population}</p>

    </article>
  )
}

Card.propTypes = {
  card: PropTypes.object
}

export default Card;