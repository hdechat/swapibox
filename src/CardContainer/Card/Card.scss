article {
  border: solid;
  height: 260px;
  margin: 0.5rem;
  width: 155px;
}