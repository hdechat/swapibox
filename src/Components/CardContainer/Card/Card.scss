article {
  border: solid;
  margin: 0.5rem;
}

h6 {
  font-size: 16px;
  margin: 1rem .5rem;
}

img {
  margin: .1rem;
  width: 15%;
}

img.active {
  background: red;
  border-radius: 50%;
}

img.not-active {
  border-radius: 50%;
}

