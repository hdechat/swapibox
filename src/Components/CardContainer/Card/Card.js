import React from 'react'
import PropTypes from 'prop-types'
import './Card.css'
import favoriteButton from '../../../assets/logo.png'
// import favorited from '../../../assets/logo-red.png'

export default function Card ({ card, clickedCard }) {
  const data = Object.entries(card).map(([key, value], index) => {
    if (key !== 'favorited') {
      if(index === Object.entries(card).length - 1) {
        return (
          <p key={value + index}>{key}: {value}</p>
        ) 
      } else {
        return (
          <div className="card-item">
            <h6 className="card-item-key" key={value + index}>{key}:</h6>
            <h6 className="card-item-value">{value}</h6>
          </div>
        )
    }
  }
})

  return (
    <article>
      <img
        onClick={() => clickedCard(!card.favorited, card)}
        className={card.favorited ? 'active' : ''}
        src={favoriteButton}
        alt="Star Wars Rebel Alliance Logo"
      />
      {data}
    </article>
  )
}

Card.propTypes = {
  card: PropTypes.object.isRequired,
  clickedCard: PropTypes.func.isRequired
}
