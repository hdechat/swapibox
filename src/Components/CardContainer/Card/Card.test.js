import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card.js'

describe('Card', () => {
  let card
  let mockCardProp = {
    name: 'Name', 
    species: 'Droid', 
    favorited: false
  }
  let mockCallback = jest.fn()

  beforeEach(() => {
    card = shallow(<Card clickedCard={mockCallback} card={mockCardProp} />)
  });

  it('matches snapshot', () => {
    expect(card).toMatchSnapshot()
  });

  it('renders the card data passed in as props', () => {
    expect(card.find('h6').length).toEqual(2)
  });

  it('calls clickedCard with the correct params when favorited image on card is clicked', () => {
    card.find('img').simulate('click')

    expect(mockCallback).toHaveBeenCalledWith(!card.favorited, mockCardProp)
  });
});
  // it('changes class to active if card.favorited is true', () => {
  //   let mockCardProp = {
  //     name: 'Name', 
  //     species: 'Droid', 
  //     favorited: true
  //   };
  //   expect(card.find('.active').length).toEqual(0);

  //   card.find('img').simulate('click')

  //   expect(card.find('.active').length).toEqual(1);
  // });