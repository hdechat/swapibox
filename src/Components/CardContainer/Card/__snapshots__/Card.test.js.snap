// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
        Object {
            "favorited": false,
            "name": "Name",
            "species": "Droid",
          }
    }
    clickedCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Star Wars Rebel Alliance Logo"
          className="not-active"
          onClick={[Function]}
          src="logo.png"
/>,
        Array [
          <h6>
            name
            : 
            Name
</h6>,
          <h6>
            species
            : 
            Droid
</h6>,
          undefined,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Star Wars Rebel Alliance Logo",
          "className": "not-active",
          "onClick": [Function],
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": "Name0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "name",
            ": ",
            "Name",
          ],
        },
        "ref": null,
        "rendered": Array [
          "name",
          ": ",
          "Name",
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": "Droid1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "species",
            ": ",
            "Droid",
          ],
        },
        "ref": null,
        "rendered": Array [
          "species",
          ": ",
          "Droid",
        ],
        "type": "h6",
      },
      undefined,
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Star Wars Rebel Alliance Logo"
            className="not-active"
            onClick={[Function]}
            src="logo.png"
/>,
          Array [
            <h6>
              name
              : 
              Name
</h6>,
            <h6>
              species
              : 
              Droid
</h6>,
            undefined,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Star Wars Rebel Alliance Logo",
            "className": "not-active",
            "onClick": [Function],
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": "Name0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "name",
              ": ",
              "Name",
            ],
          },
          "ref": null,
          "rendered": Array [
            "name",
            ": ",
            "Name",
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": "Droid1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "species",
              ": ",
              "Droid",
            ],
          },
          "ref": null,
          "rendered": Array [
            "species",
            ": ",
            "Droid",
          ],
          "type": "h6",
        },
        undefined,
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
